pkg/solana/logpoller/filters.go

```diff
@@ -228,7 +230,7 @@ func (fl *filters) removeFilterFromIndexes(filter Filter) {
 		}
 	}
 
-	discriminatorHead := filter.Discriminator()[:10]
```

Comment:
We must store only the first 10 characters (hence the name `discriminatorHead`). Storing all 12 is not useful and will cause us to miss logs. (more on this in another comment)